caracal/Dockerfile

```diff
@@ -0,0 +1,41 @@
+ARG REGISTRY
+ARG RELEASE
+
+FROM ghcr.io/vexxhost/ubuntu-cloud-archive:caracal
+# renovate: name=openstack/requirements repo=https://opendev.org/openstack/requirements.git branch=stable/2024.1
+ARG REQUIREMENTS_GIT_REF=c1cde0f897da8c355d03eacda5cf3b610fe75e2c
+ADD --keep-git-dir=true https://opendev.org/openstack/requirements.git#${REQUIREMENTS_GIT_REF} /src/requirements
+RUN cp /src/requirements/upper-constraints.txt /upper-constraints.txt
+RUN <<EOF sh -xe
+sed -i '/glance-store/d' /upper-constraints.txt
+sed -i '/horizon/d' /upper-constraints.txt
+EOF
+
+FROM ghcr.io/vexxhost/python-base:caracal
+RUN <<EOF bash -xe
+apt-get update -qq
+apt-get install -qq -y --no-install-recommends \
+    build-essential \
+    git \
+    libldap2-dev \
+    libpcre3-dev \
+    libsasl2-dev \
+    libssl-dev \
+    lsb-release \
+    openssh-client \
+    python3 \
+    python3-dev \
+    python3-pip \
+    python3-venv
+apt-get clean
+rm -rf /var/lib/apt/lists/*
+EOF
```

Comment:
Let's change this to be similar to this to include the cache:

https://github.com/vexxhost/docker-python-base/blob/main/antelope/Dockerfile#L3-L12